packages/sanity/src/core/store/events/getDocumentAtRevision.ts

```diff
@@ -0,0 +1,46 @@
+import {type SanityClient, type SanityDocument} from '@sanity/client'
+import {type Observable, of} from 'rxjs'
+import {catchError, map, startWith, tap} from 'rxjs/operators'
+
+import {type EventsStoreRevision} from './types'
+
+const documentRevisionCache: Record<string, SanityDocument> = Object.create(null)
+
+export function getDocumentAtRevision({
+  client,
+  documentId,
+  revisionId,
+}: {
+  client: SanityClient
+  documentId: string
+  revisionId: string | null | undefined
+}): Observable<EventsStoreRevision | null> {
+  if (!revisionId) {
```

Comment:
It simplifies the consumer side, because the revision is possibly undefined. But agree with you now that I'm seeing it again, it should be handled from the consumer and not from here.
Will update it, thanks!

Reply:
https://github.com/sanity-io/sanity/pull/7922/commits/1ccdfa56f47fcbb00f4ddc440c85753bf7c9e225